### PR TITLE
Addition of a test for sorting of children records in the admin API

### PR DESCRIPTION
**Pull request by VaultVulp**
This PR is almost identical to https://github.com/lektor/lektor/pull/280 , but it also includes the test that was requested by @tariquesani

### CODE REVIEW

**Comment by nixjdm** [comment]
I'm really excited by that you've picked up that work! Not only does this offer a new feature (control of the children ordering in admin) it opens up the API to further testing. I will review this soon, hopefully tomorrow.

File: tests/test_api.py [hunk 1]
```diff
@@ -0,0 +1,64 @@
...
+            )
+
+
+def test_children_sorting_via_api(scratch_project, scratch_env, children_records_data):
+    webadmin = WebAdmin(scratch_env, output_path=scratch_project.tree)
+    data = json.loads(webadmin.test_client().get('/admin/api/recordinfo?path=/myobj').data)
+    children_records_ids_provided_by_api = list(map(itemgetter('id'), data['children']))
+
+    records_ordered_by_title = sorted(children_records_data, key=itemgetter('title'))
+    ordered_records = sorted(records_ordered_by_title, key=itemgetter('pub_date'), reverse=True)
```
[the remark this answers is not in the document]
Comment (nixjdm) [reply tests/test_api.py:62]: @VaultVulp Does the test work as is for you on your machine? Can you tell me what `echo $LANG; echo $LC_ALL` outputs? I'm trying to find the same info about Travis

[the remark this answers is not in the document]
Comment (VaultVulp) [reply tests/test_api.py:62]: Works flawlessly on my MacBook Pro:
`echo $LANG; => en_US.UTF-8`
`echo $LC_ALL; =>en_US.UTF-8`

[the remark this answers is not in the document]
Comment (VaultVulp) [reply tests/test_api.py:62]: Maybe the problem lies in the OS-dependent linebreakers? For they probably can produce invalid model and content files.

[the remark this answers is not in the document]
Comment (nixjdm) [reply tests/test_api.py:62]: I'm on Linux, so we should be the same. Plus, all of the tests in test_db.py are passing for me, which are very similar. I'm still digging into it.

[the remark this answers is not in the document]
Comment (nixjdm) [reply tests/test_api.py:62]: Got it. My env was the problem. This code is fine :)

**Comment by nixjdm** [comment]
Thank you @VaultVulp !

**Pull request merged**